packages/components/pie-webc-testing/src/helpers/page-object/base-page.ts

```diff
@@ -1,10 +1,77 @@
 import { type Page } from '@playwright/test';
+import { buildUrl } from './storybook-extensions';
+
+declare global {
+    interface Window { __eventsArray: Array<string>; }
+}
 
 export class BasePage {
     readonly page: Page;
+    componentName: string;
+    componentTag: string;
+    path: string;
+    args: string;
 
-    constructor (page: Page) {
+    constructor (page: Page, componentName: string, componentTag = 'data-test-id') {
         this.page = page;
+        this.componentName = componentName;
+        this.componentTag = componentTag;
+        this.path = '';
+        this.args = '';
+    }
+
+    async load (queries: Record<string, unknown> = {}) {
+        const pageUrl = buildUrl(this.componentName, this.composePath(queries), this.args);
+        await this.open(pageUrl);
+    }
+
+    async open (url: string) {
+        await this.page.goto(url, { waitUntil: 'networkidle' });
+        return this;
+    }
+
+    composePath (queries: Record<string, unknown>) {
+        if (!queries) {
+            return '';
+        }
+
+        const flattenQueries = (obj: Record<string, unknown>, prefix = ''): string[] => Object.keys(obj).flatMap((key) => {
```

Comment:
You could use object.entries in the same way instead of object.keys and then in the flatMap pull out the value removing the need for the line below. Although that method coverts it to an array but i think that's ok in your case?